aio/aio-proxy/aio_proxy/search/es_search_runner.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from datetime import timedelta
+
+from aio_proxy.search.cache.cache import cache_strategy
+from aio_proxy.search.es_index import ElasticsearchSireneIndex
+from aio_proxy.search.geo_search import geo_search_builder
+from aio_proxy.search.text_search import text_search_builder
+
+TIME_TO_LIVE = timedelta(days=31)
+MIN_EXECUTION_TIME = 400
+MAX_TOTAL_RESULTS = 10000
+
+
+class ElasticSearchRunner:
+    def __init__(self, search_params, search_type="text"):
+        self.es_search_client = ElasticsearchSireneIndex.search()
+        self.search_type = search_type
+        self.search_params = search_params
+        self.is_text_search = False
+        self.es_search_results = None
+        self.total_results = None
+        self.execution_time = None
+        self.run()
+
+    def build_es_search_text_query(self):
```

Comment:
Not sure I understand what these redeclaration bring to the table

Reply:
Nice catch, it was an oversight!